nadlogar/problems/models.py

```diff
@@ -239,3 +239,149 @@ def generate(self):
+        pogoj_latex = {"|": r"\mid", "<": r"\lt", "<=": r"\le"}
+        return {
+            "n": sympy.latex(sympy.simplify(a * n + b)),
+            "pogoj": pogoj_latex[pogoj],
+            "stevilo": sympy.latex(stevilo),
+            "mnozica": sympy.latex(mnozica),
+        }
+
+
+class PotencnaMnozica(Problem):
+    """Problem za izpis potenčne množice od dane množice"""
+
+    class Meta:
+        verbose_name = "potenčna množica"
+
+    def generate(self):
+        velikost = random.randint(2, 3)
+        mnozice = [
+            ["a", "b", "c"],
+            [1, 2, 3],
+            ["x", "y", "z"],
+            ["alpha", "beta", "gamma"],
+            ["Pi", "Phi", "Xi"],
+            [3, 6, 9],
+            [3, 7, 42],
+        ]
+        mnozica = sympy.FiniteSet(*random.choice(mnozice)[:velikost])
+        potencna = mnozica.powerset()
+        return {"mnozica": sympy.latex(mnozica), "potencna": sympy.latex(potencna)}
+
+
+class OperacijeMnozic(Problem):
+    """Naloga za zapis unije, preseka, razlike, in kartezičnega produkta množic."""
+
+    class Meta:
+        verbose_name = "operacije z množicami"
+
+    @staticmethod
+    def generiraj_mnozico(velikost, od, do):
+        """Pripravi naključno množico dane velikosti"""
+        if od > do:
+            od, do = do, od
+        if do - od < velikost:  # poskrbimo, da je na voljo dovolj elementov
+            do = od + velikost
+        izbor = [x for x in range(od, do + 1)]
+        mnozica = sympy.FiniteSet(*random.sample(izbor, velikost))
+        return mnozica
+
+    def generate(self):
+        A = self.generiraj_mnozico(random.randint(3, 4), 1, 6)
+        B = self.generiraj_mnozico(random.randint(3, 4), 1, 6)
+        unija = A.union(B)
+        presek = A.intersection(B)
+        brez = sympy.Complement(A, B)
+        kartezicno = sympy.FiniteSet(*A * B)
+        return {
+            "A": sympy.latex(A),
+            "B": sympy.latex(B),
+            "unija": sympy.latex(unija),
+            "presek": sympy.latex(presek),
+            "brez": sympy.latex(brez),
+            "kartezicno": sympy.latex(kartezicno),
+        }
+
+
+class IzpeljaneMnozice(Problem):
+    """Problem za zapis komplementa, unije in razlike množic ter izpis elementov izpeljane množice pri podani univerzalni množici."""
+
+    class Meta:
+        verbose_name = "izpeljane množice"
+
+    def generate(self):
+        k = sympy.symbols("k")
+        a = random.randint(2, 5)
+        b = random.choice([-4, -3, -2, -1, 0, 1, 2, 3, 4])
+        c = random.randint(2, 5)
+        d = random.choice([-4, -3, -2, -1, 0, 1, 2, 3, 4])
+        if abs(b) == a or abs(d) == c:
+            raise GeneratedDataIncorrect
+        velikost_univerzalne = random.randint(12, 20)
+        univerzalna = sympy.FiniteSet(*range(1, velikost_univerzalne + 1))
+        navodilo_A = a * k + b
+        navodilo_B = c * k + d
+        A = sympy.FiniteSet(
+            *[
```

Comment:
pomoje je tukaj bolj berljivo, če se definira novo spremeljivko za izpeljan seznam, pa potem to spremenljivko razpakiras. enako spodaj.